studio.py

```diff
@@ -21,6 +21,9 @@ def cli(app, record, certutil, url, path):
         app = APPS[app](proxy, certutil)
         app.start(url)
 
+        raw_input("Press any key to capture screenshot\n")
```

Comment:
I expect this will get a bit lost as the stdout from mitmproxy is shown in the console. Typically I kill raptor studio using control+c, so I suspect a screenshot will not be taken.

Reply:
No, the screenshot is not taken on control+c.

Reply:
Rather than prompting for any key, could we make `App` a context manager, so we take a screenshot whenever the app is torn down? This should catch the ctrl+c case.

Reply:
> Rather than prompting for any key, could we make `App` a context manager, so we take a screenshot whenever the app is torn down? This should catch the ctrl+c case.

The best thing to do seems to me, for the moment, to give the mitmproxy the app object `with MITMProxy(path, record, app) as proxy` and call the screenshot method on __exit__:
`def __exit__(self, *args): 
    app.take_screenshot()
    self.process.terminate()`
I know you don't like this kind of code into mitmproxy, but the `__exit__` method is the right context to call the screenshot.

Reply:
Is there a problem with making the app object a context manager and taking the screenshot on `__exit__`? The code would then be something like:

```
with APPS[app](proxy, certutil, url) as app:
    raw_input("Press any key to quit")
```

We can then also attempt to stop the running application after taking the screenshot.

Reply:
I had the feeling that this is a deadlock, but if it works, sure.
```with MITMProxy(path, record) as proxy:
    with APPS[app](proxy, certutil, url) as app:
        raw_input("Press any key to quit")```

I also think it's a good idea to attempt to stop the running application after taking the screenshot.

Reply:
If I add an inner context to `proxy` PyCharm overwrites the `CTRL+C` with `copy` and moves the termination shortcut to `CTRL+F2`. But the code with catch it and will take the screenshot right before the termination. Also, there is the possibility to just press `ENTER` in the run window and it will terminate.